certificate-transparency does not build with protobuf 3.7.0

Latest certificate-transparency (66f725f57c5cfd3a780b1e16504d9c0946769d63) does not build with protobuf 3.7.0.
It's related to https://github.com/protocolbuffers/protobuf/issues/5869.

```
c++ -DHAVE_CONFIG_H -I. -I./cpp  -I/usr/src/gmock/include -I/usr/src/gmock/gtest/include -I/usr/local/include/evhtp   -I/usr/local/include/json-c  -I/usr/local/include/evhtp -isystem /usr/local/include -D_FORTIFY_SOURCE=2 -fno-exceptions -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing  -lgmock -lgtest -isystem /usr/local/include  -Qunused-arguments -fstack-protector-all -fPIC -Wa,--noexecstack -Wformat -Wformat-security -MT cpp/server/certificate_handler.o -MD -MP -MF $depbase.Tpo -c -o cpp/server/certificate_handler.o cpp/server/certificate_handler.cc &&\
mv -f $depbase.Tpo $depbase.Po
In file included from cpp/server/certificate_handler.cc:4:
In file included from ./cpp/server/certificate_handler.h:5:
In file included from ./cpp/log/database.h:10:
./cpp/log/logged_entry.h:12:29: error: base 'LoggedEntryPB' is marked 'final'
class LoggedEntry : private ct::LoggedEntryPB {
                            ^
./proto/ct.pb.h:2927:7: note: 'LoggedEntryPB' declared here
class LoggedEntryPB final :
      ^             ~~~~~
```

The full log can be retrieved from https://people.FreeBSD.org/~sunpoet/logs/certificate-transparency-20181212.log

--- a/DEPS
+++ b/DEPS
@@ -7,7 +7,7 @@ deps = {
      "json-c": 					 "https://github.com/AlCutter/json-c.git@json-c-0.12-20140410-fix",
      "libevent": 				 "https://github.com/libevent/libevent.git@release-2.1.10-stable",
      "libevhtp":         "https://github.com/criticalstack/libevhtp.git@1.2.18",
-     "protobuf":         "https://github.com/google/protobuf.git@v2.6.1",
+     "protobuf":         "https://github.com/google/protobuf.git@v3.8.0",
      "protobuf/gtest":   "https://github.com/google/googletest.git@release-1.7.0",
      "tcmalloc":				 "https://github.com/gperftools/gperftools.git@gperftools-2.4"
 }

